phpunit/class-wp-rest-pattern-directory-controller-test.php

```diff
@@ -68,6 +68,49 @@ public function tear_down() {
 		parent::tear_down();
 	}
 
+	/**
+	 * @covers WP_REST_Pattern_Directory_Controller::register_routes
+	 *
+	 * @since 5.8.0
+	 * @since 6.2.0 Added pattern directory categories endpoint.
+	 */
+	public function test_register_routes() {
+		$routes = rest_get_server()->get_routes();
+
+		$this->assertArrayHasKey( '/wp/v2/pattern-directory/patterns', $routes );
+		$this->assertArrayHasKey( '/wp/v2/pattern-directory/categories', $routes );
+	}
+
+	/**
+	 * @covers WP_REST_Pattern_Directory_Controller::prepare_pattern_category_for_response
+	 *
+	 * @since 6.2.0
+	 */
+	public function test_prepare_pattern_category_for_response() {
+		$raw_categories = array(
+			(object) array(
+				'id'          => 3,
+				'name'        => 'Columns',
+				'slug'        => 'columns',
+				'description' => 'A description',
+			),
+		);
+
+		$prepared_category = static::$controller->prepare_response_for_collection(
+			static::$controller->prepare_pattern_category_for_response( $raw_categories[0], new WP_REST_Request() )
+		);
+
+		$this->assertSame(
+			array(
+				'id'   => 3,
+				'name' => 'Columns',
+				'slug' => 'columns',
+			),
+			$prepared_category
+		);
+	}
```

Comment:
Maybe we should use assertions similar to the `::prepare_item_for_response` test - https://github.com/WordPress/wordpress-develop/blob/master/tests/phpunit/tests/rest-api/rest-pattern-directory-controller.php#L324-L334. What do you think?

Reply:
I don't think it makes any difference to load from a json file, since we're only checking if it preserves the props we want to and we don't make any request. That's why I also didn't use `assertArrayNotHasKey`, because it makes sense to test what to expect and not if just a `key` isn't there.